File: gamepack/Decompiled/Canvas.java
import java.awt.Graphics;

public final class Canvas extends java.awt.Canvas {
   static IndexedSprite ay;
   java.awt.Component af;

   Canvas(java.awt.Component var1) {
      this.af = var1;
   }

   public final void update(Graphics var1) {
      this.af.update(var1);
   }

   public final void paint(Graphics var1) {
      this.af.paint(var1);
   }

   public static int ac_renamed(int var0, int var1, int var2) {
      int var4 = fn.an_renamed(var2 - var1 + 1);
      var4 <<= var1;
      return var0 & ~var4;
   }

   static void ne_renamed() {
      Client.in.aw(mi.an_renamed(ClientProt.ac, Client.in.au));
      Client.lv = 0;
   }
}

File: updater-runestar/src/main/java/org/runestar/client/updater/mapper/identifiers/classes/Canvas.kt
package org.runestar.client.updater.mapper.identifiers.classes

import org.runestar.client.updater.mapper.abstractclasses.IdentityMapper
import org.runestar.client.updater.mapper.annotations.MethodParameters
import org.runestar.client.updater.mapper.predicateutilities.mark
import org.runestar.client.updater.mapper.predicateutilities.predicateOf
import org.runestar.client.updater.mapper.predicateutilities.type
import org.runestar.client.updater.mapper.wrappers.Class2
import org.runestar.client.updater.mapper.wrappers.Field2
import org.runestar.client.updater.mapper.wrappers.Method2
import java.awt.Canvas
import java.awt.Component

class Canvas : IdentityMapper.Class() {
    override val predicate = predicateOf<Class2> { it.superType == Canvas::class.type }

    class component : InstanceField() {
        override val predicate = predicateOf<Field2> { it.type == Component::class.type }
    }

    @MethodParameters("g")
    class paint : InstanceMethod() {
        override val predicate = predicateOf<Method2> { it.mark == Canvas::paint.mark }
    }

    @MethodParameters("g")
    class update : InstanceMethod() {
        override val predicate = predicateOf<Method2> { it.mark == Canvas::update.mark }
    }
}